### PR TITLE
Test Fixes for 503 error 

### PR DESCRIPTION
**Pull request by gavin-aguiar**
## Description

Fixes # https://github.com/Azure/azure-functions-pyfx-planning/issues/118

- Fixed Docker tests
- Added 3.11 for docker tests.
- Skipped extensions for http tests

---

### PR information

- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made and CI is passing.

### Quality of Code and Contribution Guidelines
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-functions-python-worker/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by hallvictoria — APPROVED** [review]

File: tests/utils/testutils.py [hunk 1]
```diff
@@ -219,54 +190,62 @@ def get_script_dir(cls):
...
         pass
 
     @classmethod
     def get_environment_variables(cls):
         pass
 
     @classmethod
-    def get_script_name(cls):
-        pass
+    def docker_tests_enabled(self) -> (bool, str):
```
Comment (YunchuWang) [inline tests/utils/testutils.py:201]: add comment whats returned?

File: tests/utils/testutils.py [hunk 2]
```diff
@@ -219,54 +190,62 @@ def get_script_dir(cls):
...
+            return True, CONSUMPTION_DOCKER_TEST
+        elif is_envvar_true(DEDICATED_DOCKER_TEST):
+            return True, DEDICATED_DOCKER_TEST
+        else:
+            return False, None
 
     @classmethod
     def setUpClass(cls):
         script_dir = pathlib.Path(cls.get_script_dir())
+        is_unit_test = True if 'unittests' in script_dir.parts else False
```
Comment (YunchuWang) [inline tests/utils/testutils.py:212]: whats this .parts checking for? inline comment?

File: tests/utils/testutils.py [hunk 3]
```diff
@@ -1046,11 +1024,11 @@ def _setup_func_app(app_root):
         with open(host_json, 'w') as f:
             f.write(HOST_JSON_TEMPLATE)
 
-    if not os.path.isfile(extensions_csproj_file):
+    if not os.path.isfile(extensions_csproj_file) and not is_unit_test:
```
Comment (YunchuWang) [inline tests/utils/testutils.py:1027]: so for unit tests, we never need extensions installed, only http tests?

File: tests/unittests/test_script_file_name.py [hunk 4]
```diff
@@ -69,12 +83,19 @@ class TestInvalidScriptFileName(testutils.WebHostTestCase):
...
-        return INVALID_SCRIPT_FILE_NAME_DIR
+    def setUpClass(cls):
+        os.environ["PYTHON_SCRIPT_FILE_NAME"] = "main"
+        super().setUpClass()
 
     @classmethod
-    def get_script_name(cls):
-        return "main"
+    def tearDownClass(cls):
+        # Remove the WEBSITE_HOSTNAME environment variable
```
Comment (YunchuWang) [inline tests/unittests/test_script_file_name.py:92]: commend is different from code

File: tests/unittests/test_script_file_name.py [hunk 5]
```diff
@@ -47,12 +54,19 @@ class TestNewScriptFileName(testutils.WebHostTestCase):
...
+    def setUpClass(cls):
+        os.environ["PYTHON_SCRIPT_FILE_NAME"] = "test.py"
+        super().setUpClass()
 
     @classmethod
-    def get_script_name(cls):
-        return "test.py"
+    def tearDownClass(cls):
+        # Remove the WEBSITE_HOSTNAME environment variable
+        os.environ.pop('PYTHON_SCRIPT_FILE_NAME')
```
Comment (YunchuWang) [inline tests/unittests/test_script_file_name.py:64]: commend differ from code

File: tests/utils/testutils.py (hunk 2 above)
Comment (gavin-aguiar) [reply tests/utils/testutils.py:212]: It checks for different paths and if any of the paths has unittests

File: tests/utils/testutils.py (hunk 3 above)
Comment (gavin-aguiar) [reply tests/utils/testutils.py:1027]: Right, we dont actually need extensions for unit tests,.

**Review by YunchuWang — APPROVED** [review]

**Pull request merged**